src/http/modules/ngx_http_upstream_zone_module.c

```diff
@@ -674,7 +679,10 @@ ngx_http_upstream_zone_init_worker(ngx_cycle_t *cycle)
                 event->log = cycle->log;
                 event->cancelable = 1;
 
-                ngx_add_timer(event, 1);
+                timer = (peer->host->valid > now
+                        ? (ngx_msec_t) 1000 * (peer->host->valid - now) : 1);
```

Comment:
Style nitpicking: either outer brackets should be removed, or second line should be indented +1 space. I'd remove the brackets. Same below.

Reply:
Another option is to use brackets only for the condition (1st line).